automation_testing/pages/DataViewPage.ts

```diff
@@ -143,4 +140,45 @@ export class DataviewPage {
 		}
 		return createdDates;
 	}
+
+	async validateContainsKeyword(titles: string[], keyword: string): Promise<void> {
+		for (const title of titles) {
+			expect(title.toLowerCase()).toContain(keyword.toLowerCase());
+		}
+	}
+
+	async validateContainsMoreThanOneKeyword(titles: string[], keywords: string[]): Promise<void> {
+
+		let isContaining = false;
+		for (const title of titles) {
+			for (const keyword of keywords) {
+				if (title.toLowerCase().includes(keyword.toLowerCase())) {
+					isContaining = true;
+				}
+			}
+			expect(isContaining, `Expected contains: '${keywords.toString()}' but was '${title}'`).toBe(true);
```

Comment:
I feel this is likely to fail on some scenarios since the "isContaining" flag is not being changed back to false on each iteration, for example let's say that titles[0] does include keywords[0] then isContaining changes to true, it executes the expect, but the flag is still true, then by next iteration, even if titles[0] does not contain keywords[1] it will still mark it as true.

I see 2 alternatives, either add an "else" that changes isContaining to false, or changing it back to false at the start of each iteration.